pkg/dataobj/builder.go

```diff
@@ -286,75 +249,47 @@ func streamSizeEstimate(stream logproto.Stream) int {
 	return size
 }
 
-// Flush flushes all buffered data to object storage. Calling Flush can result
+// Flush flushes all buffered data to the buffer provided. Calling Flush can result
```

Comment:
I think Flush being a no-op when there was nothing to flush worked when it had side effects (ie potentially uploading to S3). 

But now that it writes to an argument, a no-op flush should probably be more obvious to the caller: I think we should return an `ErrBuilderFull` if you try to flush with an empty builder.

(also ErrBufferFull should probably be ErrBuilderFull, I think that was a typo on my part)